specs/_features/ePBS/p2p-design.md

```diff
@@ -0,0 +1,65 @@
+# ePBS -- Networking Design Notes
+
+## Gossip validation considerations
+
+### Signed builder bid (aka. execution payload header with builder attribution and value)
+
+The signed builder bid arrives at slot `N` for the inclusion of `N+1`. When a node receives the builder's bid, the node will make one of the three decisions. `ACCEPT`, `IGNORE`, or `REJECT`. These are gossip sub lingo which I won't get too much into. `ACCEPT` forwards to its peers. `IGNORE` and `REJECT` both drop the object, the only difference is `REJECT` penalizes the forwarding peer. The builder bid validations consist of four components with the corresponding validation questions.
+- Time
+    - Did the builder's bid arrive at the right time?
+- Value
+    - Did the builder have enough to pay for the bid?
+- Chain extension
+    - Did the builder's bid extend a valid chain tip from the chain's perspective?
+- Attribution
+    - Did the bid come from a builder on the chain?
+- Signature
+    - Did the bid's signature validate for the builder?
+
+Let's first tackle the `time` check. A node must drop a late bid (ie `< N+1`), and can drop a future bid (ie `> N+1`), although It's not clear to me why a builder wants to send an early bid. In both cases, I think `IGNORE` makes more sense than `REJECT`, typically in gossip validations, time-bound checks all used `IGNORE`. A node could be syncing to the tip of the chain.
+
+For the `value` check. A node must drop a bid if the builder does not have enough balance to pay for the bid plus maintain a self-min balance. I think `REJECT` makes more sense here. A node should not gossip about this because it's no use.
+
+For the `chain_extension` check. Suppose a chain has multiple tips. Builders maintain the same view. Competitive builders will be building simultaneously on all the tips. One paradigm shift is builder bid cancellation is no longer viable. This differs from the current mev-boost model. A node should `IGNORE` builder bids that are not top-of-the-chain tips. Worrying about DOS, a node can accept multiple bids of the same builder on the same tip, but the value has to be incremental. We need to think about this more.
```

Comment:
How feasible it is to have a `PriceBump` field for bids similar to what we have in transaction pool? Only accept if the new bid for same tip is `x%` more than previous one.

Reply:
Not really sure if it has to be enforced by the protocol or can be made configurable. Configurable might be favourable for some proposers.